app/assets/javascripts/Components/submission_table.jsx

```diff
@@ -312,25 +311,35 @@ class SubmissionsActionBox extends React.Component {
 
   render = () => {
     let completeButton, incompleteButton, collectButton, runTestsButton, releaseMarksButton, unreleaseMarksButton;
-    if (this.props.is_admin) {
-      completeButton = (
-        <button
-          onClick={this.props.completeResults}
-          disabled={this.props.disabled}
-        >
-          {I18n.t('results.set_to_complete')}
-        </button>
-      );
 
-      incompleteButton = (
-        <button
-          onClick={this.props.incompleteResults}
-          disabled={this.props.disabled}
+    completeButton = (
+      <button
+        onClick={this.props.completeResults}
+        disabled={this.props.disabled}
+      >
+        {I18n.t('results.set_to_complete')}
+      </button>
+    );
+
+    incompleteButton = (
+      <button
+        onClick={this.props.incompleteResults}
+        disabled={this.props.disabled}
+      >
+        {I18n.t('results.set_to_incomplete')}
+      </button>
+    );
+
+    if (this.props.can_run_tests) {
+      runTestsButton = (
+        <button onClick={this.props.runTests}
+                disabled={this.props.disabled}
         >
-          {I18n.t('results.set_to_incomplete')}
+          {I18n.t('submissions.run_tests')}
         </button>
       );
-
```

Comment:
Please do not delete blank lines in existing code. Revert this change, and the one below.